src/crimson/osd/main.cc

```diff
@@ -79,26 +122,36 @@ int main(int argc, char* argv[])
   }
   std::string cluster_name;
   std::string conf_file_list;
+
   // ceph_argparse_early_args() could _exit(), while local_conf() won't ready
   // until it's started. so do the boilerplate-settings parsing here.
   auto init_params = ceph_argparse_early_args(ceph_args,
                                               CEPH_ENTITY_TYPE_OSD,
                                               &cluster_name,
                                               &conf_file_list);
+
+
   seastar::sharded<OSD> osd;
   seastar::sharded<ceph::net::SocketMessenger> cluster_msgr, client_msgr;
   seastar::sharded<ceph::net::SocketMessenger> hb_front_msgr, hb_back_msgr;
   using ceph::common::sharded_conf;
   using ceph::common::sharded_perf_coll;
   using ceph::common::local_conf;
+  ObserverMgr<ceph::md_config_obs_impl<ConfigProxy>> config_tracker;
+  ConfigValues config_values;
+  config_values.name = init_params.name;
+  config_values.cluster = cluster_name;
+  md_config_t ceph_config(config_values, config_tracker, true);
+  ceph_config.parse_config_files(config_values, config_tracker, conf_file_list.c_str(),&std::cerr, CODE_ENVIRONMENT_DAEMON);
+  ceph_config.parse_argv(config_values,config_tracker, ceph_args);
```

Comment:
the purpose of this hunk is to populate the settings to config_values. so i'd suggest extract it into its own function which returns a `unique_ptr< ConfigValues>`.

Reply:
do you mean not use md_config_t.parse_config_files?  I think we still need md_config_t, it will parse options first, we can just reuse it.

Reply:
no, i suggested to have a separated function to populate the `ConfigValues`. but anyway, since we will not do this before reactor is started, this does not matter anymore.